ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/AgentUtils.java

```diff
@@ -150,17 +159,46 @@ public static String addContextToPrompt(Map<String, String> parameters, String p
         return prompt;
     }
 
+    public static List<String> MODEL_RESPONSE_PATTERNS = List
```

Comment:
If any of the fields like thought, action, or action_input could contain nested JSON objects or arrays rather than simple strings, you might need patterns that can match these structures.

```
"\\{\\s*\"thought\"\\s*:\\s*\\{.*?\\}\\s*,\\s*\"action\"\\s*:\\s*\\{.*?\\}\\s*,\\s*\"action_input\"\\s*:\\s*\\{.*?\\}\\s*}"
```

Reply:
For a more flexible approach that can match any of the fields in any order, including the optional error field (if we expect any error in our llm output), we might consider something like this:

```
"\\{\\s*(\"(thought|action|action_input|final_answer|error)\"\\s*:\\s*\".*?\"\\s*,?\\s*)+\\}"
```

Reply:
Thanks, this part is what I planned to tune, will try your general pattern